usr/lib/common/shared_memory.c

```diff
@@ -202,6 +218,18 @@ int sm_open(const char *sm_name, int mode, void **p_addr, size_t len, int force)
         goto done;
     }
 
+    /*
+     * If the shared memory segment does not belong to the pkcs11 group or does
+     * not have correct permissions, do not use it.
+     */
+    if (stat_buf.st_gid != grp->gr_gid ||
+        (stat_buf.st_mode & ~S_IFMT) != (unsigned int)mode) {
+        TRACE_ERROR("SHM segment has wrong gid/mode combination (expected: %u/0%o; got: %u/0%o)\n",
```

Comment:
Should this message maybe also go to the Syslog? How would a user know what's wrong when this hits? 

This might be especially important for migration cases, i.e. when one migrates from a version without this change to a version with this change. As you mention in the commit message, a reboot might be required to force the re-creation of the segments.

Reply:
This is in the user program, not the pkcsslotd.  So I thought we should not put this in syslog but use the regular tracing.  At least i used it when my `pkcsconf` showed 0 slots (just remove the `& ~S_IFMT` and you will get the error).  Do we really want this in the syslog?

Reply:
Well, I would say that failures due to setup problems should probably go to syslog. We do this for other situations with setup-type errors, too. 
Example: api_interface.c:
```
        OCK_SYSLOG(LOG_ERR, "C_Initialize: Module failed to attach to "
                   "shared memory. Verify that the slot management "
                   "daemon is running, errno=%d\n", errno);
```
Other, regular, non-setup related errors should of course go to trace only.

Reply:
Changed the `TRACE_ERROR` to `OCK_SYSLOG`.

Reply:
Thanks. Just one minor thing: We usually have both, TRACE **and** SYSLOG, so that you  also see in the trace which messages were written to syslog (i.e. if you only have the trace output, but not the syslog).

Reply:
Fixed.